### DISCUSSION TITLE
MiXCR 4 equivalent of exportClones -n parameter in MiXCR 3

### DISCUSSION THREAD

**Discussion by abspangler13** [milaboratory/mixcr#1750] [Q&A] [answered]
Hello, 

We used to run the exportClones function in MiXCR 3 with the -n argument set to 3:
`mixcr exportClones -n 3 -p fullImputed input.clns output.txt`

We are currently running this as part of the `analyze` function in MiXCR 4:
```
mixcr analyze smart-seq2-vdj \
    --species hsa \
    "${R1fastq_L1}" \
    "${R2fastq_L1}" \
    "${analysisFilePath}/${each_sampleNAME}"
```
It appears to default to exporting 2 clones and I'm wondering how to make it export 3. 

Thanks,

Abby

**Comment by mizraelson** [comment]
Hi, `smart-seq2-vdj` is a single-cell protocol, which is why the number of possible clones assembled is limited to two per chain. This limitation occurs during the assembly step, not the export. Why do you expect more than two clones per chain in single-cell data?

**Reply by abspangler13** [reply]
I interpreted the number of clones in the output file to be the number of clones per well of the plate. Ideally you would see one clone per well, but in some instances multiple cells are deposited into a well or there is contamination that could result in multiple clones being sequenced for single well.

**Reply by mizraelson** [reply, marked as answer]
MiXCR v4 presets are specifically designed to address such contaminations. That’s why, for single-cell protocols, it assembles a maximum of 2 clones per chain.

However, you can try allowing more consensuses by adding the following parameter:

`-Massemble.consensusAssemblerParameters.assembler.maxConsensuses=3`

**Reply by abspangler13** [reply]
Thanks! When you say it addresses contamination, what does that mean? We look for contamination by using our negative wells and looking across the entire plate. Since MiXCR is run per well, I'm not sure how it addresses contamination.

**Reply by mizraelson** [reply]
In the case of this preset, there are two steps:

1) Only highly abundant consensuses are retained after the assembly step.
2) As part of the analyze pipeline, the `mixcr assembleCells` command is used, which compares clones present in each well across the entire plate. The wells (cells) are grouped by the clones they contain, and the results can be seen in the grouped TSV file.

MiXCR is run on all wells simultaneously, not separately. For instance, if you have a set of files like:

```
161014_SN163_0729_AH3VW7BCXY_L1_TAAGGCGA-CTCTCTAT_R1.fastq.gz
161014_SN163_0729_AH3VW7BCXY_L1_TAAGGCGA-CTCTCTAT_R2.fastq.gz
161014_SN163_0729_AH3VW7BCXY_L1_TAAGCTTA-GTGTTAAG_R1.fastq.gz
161014_SN163_0729_AH3VW7BCXY_L1_TAAGCTTA-GTGTTAAG_R2.fastq.gz
...
```

Where the first barcode in the filename defines the row and the second defines the column, you would run MiXCR as follows:

```
mixcr analyze smart-seq2-vdj \
    --species hsa \
    161014_SN163_0729_AH3VW7BCXY_L1_{{CELL0ROW:a}}-{{CELL0COL:a}}_R1.fastq.gz \
    161014_SN163_0729_AH3VW7BCXY_L1_{{CELL0ROW:a}}-{{CELL0COL:a}}_R2.fastq.gz \
    result
```
This command will aggregate all FASTQ files for the plate.
